nucleus/utils.py

```diff
@@ -422,3 +425,55 @@ def paginate_generator(
             yield json_value
         if not next_token:
             break
+
+
+def get_image_dimension(image_fpath: str) -> Tuple[int, int]:
+    im = Image.open(image_fpath)
+    return im.size
+
+
+GLOB_SIZE_THRESHOLD_CHECK = 500
+
+
+def create_items_from_folder_crawl(
+    dirname: str,
+    file_globs: Tuple[str],
+    use_privacy_mode: bool,
+    privacy_mode_proxy: str,
+    skip_size_warning: bool,
+) -> List[DatasetItem]:
+    relative_fpaths = []
+    for file_glob in file_globs:
+        pathname = f"**/*.{file_glob}"
+        print(f"Searching for {pathname} in root dir: {dirname}")
+        fpaths = glob.glob(pathname=pathname, root_dir=dirname, recursive=True)
+        relative_fpaths.extend(fpaths)
+
+        if (
+            len(relative_fpaths) > GLOB_SIZE_THRESHOLD_CHECK
+            and not skip_size_warning
+        ):
+            raise Exception(
+                f"Found over {GLOB_SIZE_THRESHOLD_CHECK} items in {dirname}. If this is intended, set skip_size_warning=True when calling this function."
+            )
```

Comment:
I can remove this, if you think its unnecessary.
Its just in case some one mis-types a folder name, and we end up globbing way more than we should.

Reply:
I think this is a fairly reasonable check to be had, I'd maybe put it into the client method instead though, that way we wouldn't need to pass the `skip_size_warning` here either

Reply:
>I'd maybe put it into the client method instead though

The benefit of having it here _inside_ the for-loop is that it will break as soon as we glob more than the threshold.
Otherwise we would have to wait for a large glob to complete, create items, and then check the threshold.